src/nsls2api/cli/environment.py

```diff
@@ -0,0 +1,140 @@
+    pass  # No need to call anything manually
+
+
+console = Console(
+    theme=Theme(
+        {
+            "info": "cyan",
+            "warning": "yellow",
+            "error": "red bold",
+            "success": "green bold",
+            "url": "blue underline",
+            "env.prod": "red bold",
+            "env.dev": "yellow",
+            "env.local": "green",
+            "env.custom": "cyan",
+        }
+    )
+)
+
+
+def create_environment_table() -> Table:
+    """Create a table showing all available environments"""
+    table = Table(box=ROUNDED, show_header=True)
+    table.add_column("Shortcut", style="cyan bold")
+    table.add_column("Environment", style="green")
+    table.add_column("URL", style="blue")
+
+    table.add_row("prod", "Production", ApiEnvironment.PRODUCTION.value)
+    table.add_row("dev", "Development", ApiEnvironment.DEVELOPMENT.value)
+    table.add_row("local", "Local", ApiEnvironment.LOCAL.value)
+
+    return table
+
+
+def get_environment_style(url: str) -> str:
+    """Get the appropriate style for the environment"""
+    if url == ApiEnvironment.PRODUCTION.value:
+        return "env.prod"
+    elif url == ApiEnvironment.DEVELOPMENT.value:
+        return "env.dev"
+    elif url == ApiEnvironment.LOCAL.value:
+        return "env.local"
+    else:
+        return "env.custom"
+
+
+def get_environment_name(url: str) -> str:
+    """Get a friendly name for the environment"""
+    if url == ApiEnvironment.PRODUCTION.value:
+        return "Production"
+    elif url == ApiEnvironment.DEVELOPMENT.value:
+        return "Development"
+    elif url == ApiEnvironment.LOCAL.value:
+        return "Local"
+    else:
+        return "Custom"
+
+
+@app.command()
+def show():
+    """Show current API environment"""
+    current_url = get_base_url()
+    env_name = get_environment_name(current_url)
+    env_style = get_environment_style(current_url)
+
+    # Create status table
+    status_table = Table(show_header=False, box=None)
+    status_table.add_column("Key", style="cyan")
+    status_table.add_column("Value")
+
+    status_table.add_row("Environment", Text(env_name, style=env_style))
+    status_table.add_row("URL", Text(current_url, style="url"))
+
+    # Create the main panel
+    panel = Panel(status_table, title="Current Environment", border_style=env_style)
+
+    # Show available environments
+    env_table = create_environment_table()
+    env_panel = Panel(env_table, title="Available Environments", border_style="cyan")
+
+    console.print(panel)
+    console.print("\n[info]Available Environments:")
+    console.print(env_panel)
+
+
+@app.command()
+def switch(
+    env: str = typer.Argument(
+        ..., help="Environment to switch to (prod/dev/local) or a custom URL"
+    ),
+):
+    """Switch API environment"""
+    env_map = {
+        "prod": ApiEnvironment.PRODUCTION.value,
+        "dev": ApiEnvironment.DEVELOPMENT.value,
+        "local": ApiEnvironment.LOCAL.value,
+    }
```

Comment:
Currently if user types "production" or "development" then a custom environment will be created.  

I wonder if it is worth adding these entries to `env_map`, or if that is a slippery slope and it is better to keep only one "correct" name for the command to respond to?

I'm unsure about which is better, but I thought it worth asking.

Reply:
No it's a good point - let me think how best to implement - but I think supporting either `dev` or `development` and `prod` or `production` is not to bad.

Reply:
This is good.  I would not have done anything differently.